File: YukariLauncher/src/main/java/net/kdt/pojavlaunch/customcontrols/mouse/TouchEventProcessor.java
package net.kdt.pojavlaunch.customcontrols.mouse;

import android.view.MotionEvent;
import android.view.View;

public interface TouchEventProcessor {
/**
 * 「process Touch Event」処理を実行します。
 * このメソッドは特定の機能を提供するために実装されています。
 */
    boolean processTouchEvent(MotionEvent motionEvent);
/**
 * 「cancel Pending Actions」処理を実行します。
 * このメソッドは特定の機能を提供するために実装されています。
 */
    void cancelPendingActions();
/**
 * タッチイベントを処理します。
 * ユーザーからのタッチ入力を検出し、適切なアクションを実行します。
 */
    default void dispatchTouchEvent(MotionEvent event, View view) {}
}

File: YukariLauncher/src/main/java/com/arata/yukarilauncher/ui/subassembly/customcontrols/mouse/TouchEventProcessor.kt
package com.arata.yukarilauncher.ui.subassembly.customcontrols.mouse

import android.view.MotionEvent
import android.view.View

/** タッチイベントを処理するプロセッサーのインターフェース。ゲーム内/メニュー画面で異なる実装を持ちます。 */
interface TouchEventProcessor {
    /**
     * タッチイベントを処理します。
     * @param motionEvent 処理するタッチイベント
     * @return イベントを消費した場合はtrue
     */
    fun processTouchEvent(motionEvent: MotionEvent): Boolean
    /** 保留中のアクションをキャンセルします。 */
    fun cancelPendingActions()
    /**
     * タッチイベントをディスパッチします。
     * @param event ディスパッチするイベント
     * @param view 対象のビュー
     */
    fun dispatchTouchEvent(event: MotionEvent, view: View) {}
}
